### ISSUE TITLE
nixos: all installer tests are broken

### ISSUE THREAD

**Issue by xeji** [NixOS/nixpkgs#45866]
All nixos installer tests are [currently broken on hydra](https://hydra.nixos.org/jobset/nixos/trunk-combined/jobs-tab?filter=nixos.tests.installer). 

[Edit] The failure is locally reproducible. It occurs when the test is already running in a VM and the installer builds the system to be installed on a virtual disk.

Apparently the installer tries to rebuild `stdenv` in the sandbox:

```
warning: unable to download 'https://rt.perl.org/Public/Ticket/Attachment/1502646/807252/0001-Fix-missing-build-dependency-for-pods.patch': Timeout was reached (28); retrying in 262 ms
warning: unable to download 'https://rt.perl.org/Public/Ticket/Attachment/1502646/807252/0001-Fix-missing-build-dependency-for-pods.patch': Timeout was reached (28); retrying in 543 ms
warning: unable to download 'https://rt.perl.org/Public/Ticket/Attachment/1502646/807252/0001-Fix-missing-build-dependency-for-pods.patch': Timeout was reached (28); retrying in 1269 ms
warning: unable to download 'https://rt.perl.org/Public/Ticket/Attachment/1502646/807252/0001-Fix-missing-build-dependency-for-pods.patch': Couldn't resolve host name (6); retrying in 2215 ms
builder for '/nix/store/4x3famhx2zy5q0khadzdj2wsp3glg6n3-0001-Fix-missing-build-dependency-for-pods.patch.drv' failed with exit code 1; last 1 log lines:
machine#   error: unable to download 'https://rt.perl.org/Public/Ticket/Attachment/1502646/807252/0001-Fix-missing-build-dependency-for-pods.patch': Couldn't resolve host name (6)
cannot build derivation '/nix/store/xqmmzan483v2kd2kf0gygi8a0r4xi4lj-perl-5.28.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/af3m2fc3akhrxplr64hwca6f2qd0dm00-bison-3.0.5.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/3lrh7i8fmc50l9qnglchkscbry2pncsg-binutils-2.30.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/8cv2zwkn59hk33ajpfnzkn8dx74n9pp0-binutils-wrapper-2.30.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/vh10p6acr82b3kkvl3ijqllvcj7a7ykx-binutils-wrapper-2.30.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/hrg6w6zdiyrw39z1vy1ijn3xbh4aqak1-bootstrap-stage2-gcc-wrapper.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/x1xc6x3lb6yyis2mxfb7fhykbl23hpsr-bootstrap-stage3-gcc-wrapper.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/c9ncq80i7xdh84gr4brsd55747cfp9v2-bootstrap-stage4-gcc-wrapper-7.3.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/b7mxlm1mc8c455ks6lh7l6i1p3hlkmh4-bootstrap-stage3-stdenv-linux.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/hla4km7vmzy7hbzy8214x3slk5n34bk3-bootstrap-stage4-stdenv-linux.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/80aia7jkgvg364mr6b0w9v468z0j5bnd-pcre-8.42.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/n55v46dmpf3gv0kni6l9ypsir8wb18lk-xz-5.2.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/cfd8kwpjak4yxs6jq20an627wh6rg864-zlib-1.2.11.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/wvnxjn0ckxvg59wqfihc17gbxxqhzfrl-system-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/d3spqdifzl3r01g63ir2p8ss25mz0x8b-unit-nixos-upgrade.service.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/h57kx48yqgdf5x0bcqmw92qb6casxqbn-nixos-system-nixos-18.09pre-git.drv': 1 dependencies couldn't be built
```

Relevant changeset :   [da11a26..16b3217](https://hydra.nixos.org/api/scmdiff?branch=&type=git&uri=https%3A%2F%2Fgithub.com%2FNixOS%2Fnixpkgs.git&rev2=16b3217148b02ad5a2149b500dbfcfa08d7b46d7&rev1=da11a26008818b8bf64aa13709a1ee6b13fc8eba)

Looks like the test itself wasn't changed.

**Comment by xeji** [comment]
cc @vcunat @samueldr

**Comment by volth** [comment]
This should fall back to https://tarballs.nixos.org/sha256/1bb4mldfp8kq1scv480wm64n2jdsqa3ar46cjp1mjpby8h5dr2r0 (which is active), but it seems that ```fetchurlBoot``` does not do that fallback

**Comment by xeji** [comment]
The tests are designed to run in a sandbox and shouldn't use the network at all.

**Comment by volth** [comment]
> The tests are designed to run in a sandbox and shouldn't use the network at all.

The failure is when ```perl``` is building, so it is before the test running, isn't it?

**Comment by xeji** [comment]
No, it's while the rest is already running in the VM and builds the system to be installed.

**Comment by dezgeg** [comment]
It's 1b11fdd0df2254a762a16d04e603ba99e42169c0 or the commit right before that. I think they introduce new build-time dependencies that are not included in the VM.

**Comment by xeji** [comment]
Thanks for the hint, working on a fix.

**Comment by xeji** [comment]
Fixed in 43e30b1eadac5f86fe6f91b13ebcace88bf8a597
